fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppAction.kt

```diff
@@ -262,4 +262,11 @@ sealed class AppAction : Action {
             val expanded: Boolean,
         ) : ShoppingAction()
     }
+
+    /**
+     * Action used to update the visibility of the Debug Drawer trigger.
+     *
+     * @property enabled Whether the debug drawer has been enabled as part of this Action.
+     */
+    data class UpdateDebugDrawerEnabled(val enabled: Boolean) : AppAction()
```

Comment:
If we aren't keeping the enabled state in the Store, I'm not sure if we need to use an action/middleware to control the side-effects

Reply:
That's fair. Since this doesn't need any other side effects, the Action/Middleware is a little heavy handed.